ampd/src/config.rs

```diff
@@ -46,6 +48,27 @@ impl Default for Config {
     }
 }
 
+impl Debug for Config {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let redacted = "redacted".to_string();
+        f.debug_struct("Config")
+            .field("tm_jsonrpc", &redacted)
+            .field("tm_grpc", &redacted)
+            .field("tm_grpc_timeout", &self.tm_grpc_timeout)
+            .field("broadcast", &self.broadcast)
+            .field("handlers", &self.handlers)
+            .field("tofnd_config", &redacted)
+            .field("event_processor", &self.event_processor)
+            .field("service_registry", &self.service_registry)
+            .field("rewards", &self.rewards)
+            .field("health_check_bind_addr", &redacted)
+            // fmt::Debug is already redacted for field gprc
+            // (@see: src/grpc/mod.rs)
+            .field("grpc", &self.grpc)
+            .finish()
+    }
+}
```

Comment:
would it make sense to bake the redaction into the types that you're redacting instead?

Reply:
Yes, let's work towards this in an another PR